src/middleware.js

```diff
@@ -47,5 +47,5 @@ export const createOfflineMiddleware = (config: Config) => (store: any) => (
     send(offlineAction, store.dispatch, config, offline.retryCount);
   }
 
-  return promise || result;
+  return config.returnPromises ? promise || result : result;
```

Comment:
If you use `await` in your code, it doesn't matter if you return a Promise or a value. Other thing is, if you are doing this feature opt-in to preserve backwards compatibility...

Reply:
I agree. This is for backwards compatibility for as long as we are treating the feature as experimental.

Reply:
Maybe open a new issue to take this in account?

Reply:
Good idea.